File: src/main/java/pl/mg/tasks/manager/model/Task.java
package pl.mg.tasks.manager.model;

import lombok.*;

import javax.persistence.*;
import java.sql.Date;
import java.util.List;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "tasks")
@EqualsAndHashCode
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Task {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Integer id;

    @Column(name = "task_name")
    private String name;

    @Column(name = "task_date")
    private Date date;

    @OneToMany(mappedBy = "task", fetch = FetchType.EAGER, cascade = CascadeType.REMOVE)
    private List<Element> elements;
}

File: frontend/src/app/model/task.ts
import {Element} from './element';

export class Task {

  public _id: number;
  public _name: string;
  public _date: string;
  public _elements: Element[];

  constructor(id: number, name: string, date: string, elements: Element[]) {
    this._id = id;
    this._name = name;
    this._date = date;
    this._elements = elements;
  }

  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get date(): string {
    return this._date;
  }

  get elements(): Element[] {
    return this._elements;
  }
}
